src/MedTrackerAPI/Features/Devices/GetDevice.cs

```diff
@@ -0,0 +1,45 @@
+using MediatR;
+using MedTrackerAPI.Endpoints;
+using MedTrackerAPI.Infrastructure;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+
+namespace MedTrackerAPI.Features.Devices;
+
+public static class GetDevice
+{
+    public record Response(int DeviceId, string Description, string Manufacturer, string Model, string PartNumber, string LotNumber);
+    
+    public sealed class Endpoint : IEndpoint
+    {
+        public void MapEndpoint(IEndpointRouteBuilder app)
+        {
+            app.MapGet("devices/{deviceId}", async (IMediator mediator, int deviceId) =>
+                {
+                    var query = new GetDeviceQuery { DeviceId = deviceId };
+                    var device = await mediator.Send(query);
+                    
+                    return device is null ? Results.BadRequest("Device not found") : Results.Ok(device);
+                }).WithTags("Devices");
+        }
+    }
+    
+    public class GetDeviceQuery : IRequest<Response>
+    {
+        public int DeviceId { get; set; }
+    }
+    
+    public class GetDeviceQueryHandler(MedTrackerDbContext context) : IRequestHandler<GetDeviceQuery, Response>
+    {
+        public async Task<Response> Handle(GetDeviceQuery request, CancellationToken cancellationToken)
+        { 
+            var device = await context.Devices.Where(d => d.Id == request.DeviceId).SingleOrDefaultAsync(cancellationToken);
```

Comment:
If you're doing a PK lookup, you can use `FindAsync()`. Passing the cancellationToken is kind of wonky to this method, but here's what you're looking for. 

`var device = await context.Devices.FindAsync([id], cancellationToken)`

If you're wanting to stick with `SingleOrDefaultAsync`, you can elide the `.Where()` and pass it to `.SingleOrDefaultAsync()` like `.SingleOrDefaultAsync(d => d.Id == request.Id, cancellationToken)`

If you're willing to bring in a library, AutoMapper is a good package for doing projection for queries that are simple like this. It has some benefits for testing, too. I'm being reductive, but here's an example of how AutoMapper + EF would do it

```
return context.Devices
    .ProjectTo<Response>(mapper.ConfigurationProvider)
    .SingleOrDefaultAsync(d => d.Id == query.Id);
```

Reply:
I didn't put in AutoMapper but that's mostly because I don't fully understand how to or why to implement it. I might have you ELI5 sometime with that one.